### PR TITLE
Support for tracing webflux-6.x and gateway-4.x

### PR DESCRIPTION
**Pull request by CzyerChen**
### Add support for spring-cloud-gateway 4.x 
- [x] Add a unit test to verify that the fix works.
- [x] Explain briefly why the bug exists and how to fix it.

why:
spring-cloud-starter-gateway 4.0.0 depends on  spring-boot-starter-webflux 3.0.0
spring-boot-starter-webflux 3.0.0  depends on spring-webflux 6.0.2
spring-webflux-5.x plugin can not trace the request
how:
add support for spring-webflux-6.x diff from 5.x by witnessmethod
upgrade spring-cloud-gateway to 4.x with spring-webflux-6.x diff from 3.x by witnessclass

### Add an agent plugin to support spring-cloud-gateway 4.x 
- [x] Add a test case for the new plugin, refer to [the doc](https://github.com/apache/skywalking-java/blob/main/docs/en/setup/service-agent/java-agent/Plugin-test.md)

- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking-java/blob/main/CHANGES.md).

### CODE REVIEW

File: test/plugin/scenarios/gateway-4.x-scenario/gateway-projectA-scenario/pom.xml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    <pluginRepositories>
+        <pluginRepository>
+            <id>spring-snapshots</id>
+            <url>https://repo.spring.io/snapshot</url>
+        </pluginRepository>
+        <pluginRepository>
+            <id>spring-milestones</id>
+            <url>https://repo.spring.io/milestone</url>
+        </pluginRepository>
+    </pluginRepositories>
```
Comment (wu-sheng) [inline test/plugin/scenarios/gateway-4.x-scenario/gateway-projectA-scenario/pom.xml:66]: I think we don't need this. Maven central should all versions?

File: test/plugin/scenarios/gateway-4.x-scenario/config/expectedData.yaml [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+segmentItems:
+- serviceName: gateway-projectB-scenario
+  segmentSize: nq 0
+  segments:
+  - segmentId: not null
+    spans:
+    - operationName: /provider/timeout/error
```
Comment (wu-sheng) [inline test/plugin/scenarios/gateway-4.x-scenario/config/expectedData.yaml:22]: Is this timeout built intentionally?

File: test/plugin/scenarios/gateway-4.x-scenario/config/expectedData.yaml [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+      - {parentEndpoint: SpringCloudGateway/RoutingFilter, networkAddress: 'localhost:18070',
+        refType: CrossProcess, parentSpanId: 1, parentTraceSegmentId: not null, parentServiceInstance: not
+          null, parentService: not null, traceId: not null}
+      skipAnalysis: 'false'
+- serviceName: gateway-projectA-scenario
+  segmentSize: nq 0
+  segments:
+  - segmentId: not null
+    spans:
+    - operationName: /provider/timeout/error
```
Comment (wu-sheng) [inline test/plugin/scenarios/gateway-4.x-scenario/config/expectedData.yaml:66]: Why is a timeout for entry? How an entry span timeout?

File: docs/en/setup/service-agent/java-agent/Supported-list.md [hunk 4]
```diff
@@ -15,7 +15,7 @@ metrics based on the tracing data.
   * Resin 3 (Optional¹)
   * Resin 4 (Optional¹)
   * [Jetty Server](http://www.eclipse.org/jetty/) 9.x -> 11.x
-  * [Spring WebFlux](https://docs.spring.io/spring/docs/current/spring-framework-reference/web-reactive.html) 5.x (Optional¹)
+  * [Spring WebFlux](https://docs.spring.io/spring/docs/current/spring-framework-reference/web-reactive.html) 5.x (Optional¹) -> 6.x
```
Comment (wu-sheng) [inline docs/en/setup/service-agent/java-agent/Supported-list.md:18]:
```suggestion
  * [Spring WebFlux](https://docs.spring.io/spring/docs/current/spring-framework-reference/web-reactive.html) 5.x (Optional¹) -> 6.x (Optional¹)
```

Comment (wu-sheng) [reply docs/en/setup/service-agent/java-agent/Supported-list.md:18]: I think `[Spring Cloud Gateway](https://spring.io/projects/spring-cloud-gateway) 2.0.2.RELEASE -> 3.x (Optional²)` is not updated.

Comment (CzyerChen) [reply docs/en/setup/service-agent/java-agent/Supported-list.md:18]: got it.

File: test/plugin/scenarios/gateway-4.x-scenario/config/expectedData.yaml (hunk 2 above)
Comment (CzyerChen) [reply test/plugin/scenarios/gateway-4.x-scenario/config/expectedData.yaml:22]: in order to check error case.
image

File: test/plugin/scenarios/gateway-4.x-scenario/gateway-projectA-scenario/pom.xml (hunk 1 above)
Comment (CzyerChen) [reply test/plugin/scenarios/gateway-4.x-scenario/gateway-projectA-scenario/pom.xml:66]: got it.

File: test/plugin/scenarios/gateway-4.x-scenario/config/expectedData.yaml (hunk 2 above)
Comment (wu-sheng) [reply test/plugin/scenarios/gateway-4.x-scenario/config/expectedData.yaml:22]: My question is, are the codes causing the timeout on testing purpose?

**Comment by wu-sheng** [comment]
Could you confirm this https://github.com/apache/skywalking-java/pull/661#discussion_r1433534466? If the codes don't seem to trigger error, we should not have them in the expected data.

File: test/plugin/scenarios/gateway-4.x-scenario/config/expectedData.yaml (hunk 2 above)
Comment (CzyerChen) [reply test/plugin/scenarios/gateway-4.x-scenario/config/expectedData.yaml:22]: according to gateway3.x, in order to check request timeout case, so add the testcasse, [issue](https://github.com/apache/skywalking-java/pull/175)

**Review by wu-sheng — APPROVED** [review]
Thanks for adding this.

**Pull request merged**